swiftide-integrations/src/duckdb/mod.rs

```diff
@@ -24,7 +24,9 @@ const DEFAULT_UPSERT_QUERY: &str = include_str!("upsert.sql");
 pub struct Duckdb {
     /// The connection to the database
     #[builder(setter(custom))]
-    connection: Arc<Mutex<duckdb::Connection>>, // should be rwlock, however, internally duckdb uses a refcell, so we need the mutex for sync :(
+    connection: Arc<Mutex<duckdb::Connection>>, /* should be rwlock, however, internally duckdb
+                                                 * uses a refcell, so we need the mutex for sync
+                                                 * :( */
```

Comment:
There's a few places like this which are manually cleaned up in the 3rd commit

Reply:
Note - RwLock is often not actually better than using a Mutex, and also the std mutex should often be preferred over the tokio one https://tokio.rs/tokio/tutorial/shared-state#on-using-stdsyncmutex-and-tokiosyncmutex (PR to fix this is coming)

Reply:
> Note - RwLock is often not actually better than using a Mutex, and also the std mutex should often be preferred over the tokio one https://tokio.rs/tokio/tutorial/shared-state#on-using-stdsyncmutex-and-tokiosyncmutex (PR to fix this is coming)

Shouldn't it be the preferred choice if you want to read in parallel and write locked?

Reply:
Yes. I'm probably misrecalling the exact discussion about this.